include/mc_rtc/log/Logger.h

```diff
@@ -134,16 +207,45 @@ struct MC_RTC_UTILS_DLLAPI Logger
    */
   void removeLogEntry(const std::string & name);
 
+  /** Remove all log entries from a given source
+   *
+   * This has no effect if no log entry is associated to the given source.
+   *
+   * \param source Source whose entries should be removed
+   *
+   */
+  void removeLogEntries(const void * source);
+
   /** Return the time elapsed since the controller start */
   double t() const;
 
+  /** Access the file opened by this Logger
+   *
+   * \note This is empty before \ref start has been called
+   */
+  const std::string & path() const;
+
+  /** Flush the log data to disk (only implemented in the synchronous method) */
+  void flush();
+
 private:
+  /** Hold information about a log entry stored in this instance */
+  struct LogEntry
+  {
+    /** What is the data source (can be nullptr) */
+    const void * source;
+    /** Callback to log data */
+    serialize_fn log_cb;
+  };
   /** Store implementation detail related to the logging policy */
   std::shared_ptr<LoggerImpl> impl_ = nullptr;
   /** Set to true when log entries are added or removed */
   bool log_entries_changed_ = false;
-  /** Contains all the log entries callback */
-  std::unordered_map<std::string, serialize_fn> log_entries_;
+  /** Contains all the log entries */
+  std::unordered_map<std::string, LogEntry> log_entries_;
 };
 
+/** Helper to log members or methods */
+#define MC_RTC_LOG_HELPER(LOGGER, NAME, SOURCE, POINTER) LOGGER.addLogEntry<decltype(POINTER), POINTER>(NAME, SOURCE)
```

Comment:
Can't we have
```cpp
#define MC_RTC_LOG_HELPER(LOGGER, NAME, SOURCE, POINTER) \
  LOGGER.addLogEntry<decltype(&SOURCE::POINTER), &SOURCE::POINTER>(NAME, SOURCE)
```

such that we could write:
```cpp
MC_RTC_LOG_HELPER(logger, "entry", this, member_)
```
instead of
```cpp
MC_RTC_LOG_HELPER(logger, "entry", this, &decltype(this)::member_)
```

would result

that would generate
```
logger.addLogEntry<decltype(&Source::method), &Source::method>("name", &source);
```